UMA-Bots/Wallet-Monitoring/src/agent.spec.ts

```diff
@@ -0,0 +1,115 @@
+    monitoredList: [TEST_MONITORED_ADDRESS],
+  },
+};
+const networkManagerTest = new NetworkManager(MOCK_NM_DATA, 0);
+const passParams = (monitoredAddress: boolean) => {
+  return [
+    "120",
+    "100",
+    "100",
+    "42161",
+    "1",
+    "1",
+    "2",
+    "2",
+    "2",
+    "3215",
+    RANDOM_ADDRESSES[0],
+    RANDOM_ADDRESSES[1],
+    monitoredAddress ? TEST_MONITORED_ADDRESS : RANDOM_ADDRESSES[2],
+    RANDOM_ADDRESSES[3],
+    false,
+  ];
+};
+
+const expectedFinding = getFindingInstance(
+  "120",
+  "1",
+  "1",
+  TEST_MONITORED_ADDRESS,
+  RANDOM_ADDRESSES[3],
+  "false"
+);
+
+describe("Monitored Wallet Usage detection bot", () => {
+  let handleTransaction: HandleTransaction = provideHandleTransaction(
+    FILLED_RELAY_EVENT,
+    networkManagerTest
+  );
+
+  it("returns empty findings if there is no even emitted", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent();
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a monitored wallet usage is detected on a non-Across contract address", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[1])
+      .addEventLog(FILLED_RELAY_EVENT, RANDOM_ADDRESSES[0], passParams(true));
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns a finding if a monitored address uses the pool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true));
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([expectedFinding]);
+  });
+
+  it("doesn't return a finding if an irrelevant event is emitted from the SpokePool", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(RANDOM_EVENT_ABI, TEST_HUBPOOL_ADDR, [
+        RANDOM_ADDRESSES[0],
+        "120",
+      ]);
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("doesn't return a finding if a non-monitored address uses the bridge", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(false));
+    const findings = await handleTransaction(txEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns N findings for N events when monitorew wallets transacted (N>=1)", async () => {
+    const txEvent: TransactionEvent = new TestTransactionEvent()
+      .setFrom(RANDOM_ADDRESSES[0])
+      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
+      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
```

Comment:
```suggestion
      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
```
```suggestion
      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
      .addEventLog(FILLED_RELAY_EVENT, TEST_HUBPOOL_ADDR, passParams(true))
```
I suggest separating these two events, so that it looks like more realistic.

Reply:
I added this in a new test. Please check if I understood you correctly